src/cart/crt_swim.c

```diff
@@ -490,6 +501,8 @@ int crt_swim_init(int crt_ctx_idx)
 		D_ERROR("crt_register_progress_cb() failed=%d\n", rc);
 		D_GOTO(cleanup, rc);
 	}
+
+	crt_swim_rpc_timeout = crt_swim_rpc_timeout_default();
```

Comment:
shouldn't we do this before we register progress cb? its possible progress can be happening in parallel and swim functions invoked before the timeout is set

Reply:
This is mostly impossible but theoretically yes. Probably it would be better to move it apper.